### PR TITLE
Add benchmark for simple send/receive

### PR DESCRIPTION
**Pull request by jhendrixMSFT**

### CODE REVIEW

File: benchmark_test.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+	}
+	defer client.Close()
+
+	session, err := client.NewSession()
+	if err != nil {
+		b.Fatal(err)
+	}
+
+	// add a random suffix to the link name so the test broker always creates a new node
+	targetName := fmt.Sprintf("BencmarkSimple %d", rand.Uint64())
```
Comment (richardpark-msft) [inline benchmark_test.go:29]:
```suggestion
	targetName := fmt.Sprintf("BenchmarkSimple %d", rand.Uint64())
```

File: benchmark_test.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+	receiver, err := session.NewReceiver(
+		amqp.LinkSourceAddress(targetName),
+	)
+	if err != nil {
+		b.Fatal(err)
+	}
+
+	msg := amqp.NewMessage([]byte("test message"))
+	for i := 0; i < b.N; i++ {
+		// simple send and receive message, no concurrency
```
Comment (richardpark-msft) [inline benchmark_test.go:47]: Does the test AMQP broker we have support a bit more than this? Ie, can we query after we've accepted the message and see the state?

Mostly just wondering how we verify correctness for our library (along with the speed).

**Review by richardpark-msft — APPROVED** [review]
Looks good to me, only minor things or questions.

File: benchmark_test.go (hunk 1 above)
Comment (jhendrixMSFT) [reply benchmark_test.go]: Fixed

File: benchmark_test.go (hunk 2 above)
Comment (jhendrixMSFT) [reply benchmark_test.go:47]: Not sure.  Is there an example API you're looking for?

**Pull request merged**